pkg/controller/factory/factory.go

```diff
@@ -122,6 +124,13 @@ func (f *Factory) WithSyncContext(ctx SyncContext) *Factory {
 	return f
 }
 
+// WithSyncDegradedOnError encapsulate the controller sync() function, so when this function return an error, the operator client
+// is used to set the degraded condition to (eg. "ControllerFooDegraded"). The degraded condition name is set based on the controller name.
+func (f *Factory) WithSyncDegradedOnError(operatorClient operatorv1helpers.OperatorClient) *Factory {
+	f.operatorClient = operatorClient
```

Comment:
rename operatorClient to something SyncDegradedOnError specific (this var drives this feature, so it should be named accordingly).

Reply:
renamed.